### PR TITLE
CORTX-32739: Automated EC2 build and deploy pipeline is getting aborted.

### PR DESCRIPTION
**Pull request by Abhirup-A-G**
Signed-off-by: Abhirup Ghosh <abhirup.ghosh@seagate.com>

# Problem Statement
- Job was getting aborted if the total build time exceed 4 hours and in post build action [ Destroy EC2] was failing from some internal error without retrying again. Refer: https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/Community/job/Community-deploy/12/

# Design
-  For Bug: Added the retry option in post build action for destroying the EC2 instance properly.
-  For Feature: Increased the timeout of the script to 6 hours 

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/AbhirupSpace/job/community_build_job2_/42/

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/deployment/community/cortx-community-deploy.groovy [hunk 1]
```diff
@@ -155,12 +155,14 @@ pipeline {
     }
     post {
         always {
+            retry(count: 3) {
```
Comment (shailesh-vaidya) [inline jenkins/deployment/community/cortx-community-deploy.groovy:158]: Can we check if we can use `terraform` retry option

File: jenkins/deployment/community/cortx-community-deploy.groovy [hunk 2]
```diff
@@ -79,7 +79,7 @@ pipeline {
                         AWS_IP=$(terraform show -json terraform.tfstate | jq .values.outputs.cortx_deploy_ip_addr.value 2>&1 | tee ip.txt)
                         IP=$(cat ip.txt | tr -d '""')                        
                         ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@${IP} sudo bash /home/centos/setup.sh
-                        sleep 120
+                        sleep 240
```
Comment (shailesh-vaidya) [inline jenkins/deployment/community/cortx-community-deploy.groovy:82]: Can we add some status check instead of arbitrary sleep

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: jenkins/deployment/community/cortx-community-deploy.groovy (hunk 2 above)
Comment (Abhirup-A-G) [reply jenkins/deployment/community/cortx-community-deploy.groovy:82]: After running the steup.sh the instance reboot actually  and failing the step. So the logics are getting bypassed.

File: jenkins/deployment/community/cortx-community-deploy.groovy (hunk 1 above)
Comment (Abhirup-A-G) [reply jenkins/deployment/community/cortx-community-deploy.groovy:158]: I tried an options but didn't work

Comment (Abhirup-A-G) [reply jenkins/deployment/community/cortx-community-deploy.groovy:158]: Hence going with the current solution

**Review by shailesh-vaidya — APPROVED** [review]

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**